app/src/main/java/org/oppia/app/player/audio/AudioViewModel.kt

```diff
@@ -1,17 +1,118 @@
+
+  val durationLiveData: LiveData<Int> by lazy {
+    processDurationLiveData()
+  }
+  val positionLiveData: LiveData<Int> by lazy {
+    processPositionLiveData()
+  }
+  val playStatusLiveData: LiveData<AudioPlayStatus> by lazy {
+    processPlayStatusLiveData()
+  }
+
+  fun setVoiceoverMappings(map : Map<String, Voiceover>) {
+    voiceoverMap = map
+  }
+
+  fun setExplorationId(id: String) {
+    explorationId = id
+  }
+
+  /** Sets language code for data binding and changes data source to correct audio */
   fun setAudioLanguageCode(languageCode: String) {
     currentLanguageCode.set(languageCode)
+    audioPlayerController.changeDataSource(voiceOverToUri(voiceoverMap[languageCode]))
+  }
+
+  /** Plays or pauses AudioController depending on passed in state */
+  fun handlePlayPause(type: AudioPlayStatus?) {
+    when (type) {
+      AudioPlayStatus.PREPARED -> audioPlayerController.play()
+      AudioPlayStatus.PLAYING -> audioPlayerController.pause()
+      AudioPlayStatus.PAUSED -> audioPlayerController.play()
+      AudioPlayStatus.COMPLETED -> audioPlayerController.play()
+      else -> {}
+    }
+  }
+
+  fun handleSeekTo(position: Int) = audioPlayerController.seekTo(position)
+  fun handleRelease() = audioPlayerController.releaseMediaPlayer()
+
+  private val playProgressResultLiveData: LiveData<AsyncResult<PlayProgress>> by lazy {
+    audioPlayerController.initializeMediaPlayer()
+  }
+
+  private fun processDurationLiveData(): LiveData<Int> {
+    return Transformations.map(playProgressResultLiveData, ::processDurationResultLiveData)
+  }
+
+  private fun processPositionLiveData(): LiveData<Int> {
+    return Transformations.map(playProgressResultLiveData, ::processPositionResultLiveData)
+  }
+
+  private fun processPlayStatusLiveData(): LiveData<AudioPlayStatus> {
+    return Transformations.map(playProgressResultLiveData, ::processPlayStatusResultLiveData)
+  }
+
+  private fun processDurationResultLiveData(playProgressResult: AsyncResult<PlayProgress>): Int {
+    if (playProgressResult.isPending()) return 0
+    return playProgressResult.getOrThrow().duration
+  }
+
+  private fun processPositionResultLiveData(playProgressResult: AsyncResult<PlayProgress>): Int {
+    if (playProgressResult.isPending()) return 0
+    return playProgressResult.getOrThrow().position
+  }
+
+  private fun processPlayStatusResultLiveData(playProgressResult: AsyncResult<PlayProgress>): AudioPlayStatus {
+    if (playProgressResult.isPending()) return AudioPlayStatus.LOADING
+    return when (playProgressResult.getOrThrow().type) {
+      PlayStatus.PREPARED -> AudioPlayStatus.PREPARED
+      PlayStatus.PLAYING -> AudioPlayStatus.PLAYING
+      PlayStatus.PAUSED -> AudioPlayStatus.PAUSED
+      PlayStatus.COMPLETED -> AudioPlayStatus.COMPLETED
+    }
+  }
+
+  private fun voiceOverToUri(voiceover: Voiceover?): String {
+    /**
+     * TODO Ask Sean how to get GCS_RESOURCE_BUCKET_NAME and if it is constant
```

Comment:
Can you send an email to Ankit to ask about this? Let's try to get this unblocked ASAP since we should resolve this before checking in the code (otherwise it won't work). @oppia/dev-workflow-team & @oppia/core-maintainers in case anyone knows where this value comes from (I couldn't find it in oppia/oppia).